test/gateway/connectionInit-extension.js

```diff
@@ -0,0 +1,327 @@
+            payload: {
+              notificationAdded: notification
+            }
+          })
+        }
+      },
+      Subscription: {
+        notificationAdded: {
+          subscribe: (root, args, { pubsub, user }) => {
+            t.equal(user, userContext)
+            return pubsub.subscribe('NOTIFICATION_ADDED')
+          }
+        }
+      }
+    },
+    federationMetadata: true,
+    subscription: { onConnect }
+  })
+
+  Promise.all([
+    userService.ready(),
+    notificationService.ready(),
+    userService.listen(0),
+    notificationService.listen(0)
+  ]).then(() => {
+    gateway.register(GQL, {
+      subscription: true,
+      gateway: {
+        services: [
+          {
+            name: 'user',
+            url: `http://localhost:${(userService.server.address()).port}/graphql`,
+            wsUrl: `ws://localhost:${(userService.server.address()).port}/graphql`,
+            wsConnectionParams: {
+              connectionInitPayload () {
+                return {
+                  headers: {
+                    authorize: true,
+                    from: 'gateway'
+                  }
+                }
+              }
+            }
+          },
+          {
+            name: 'notification',
+            url: `http://localhost:${(notificationService.server.address()).port}/graphql`,
+            wsUrl: `ws://localhost:${(notificationService.server.address()).port}/graphql`,
+            wsConnectionParams: {
+              connectionInitPayload () {
+                return {
+                  headers: {
+                    authorize: true,
+                    from: 'gateway'
+                  }
+                }
+              }
+            }
+          }
+        ]
+      }
+    })
+
+    gateway.listen(0, err => {
+      t.error(err)
+      async function addUser () {
+        await gateway.inject({
+          method: 'POST',
+          url: '/graphql',
+          body: {
+            query: `
+              mutation {
+                addUser(name: "titi") {
+                  id
+                }
+              }
+            `
+          }
+        })
+      }
+
+      async function addNotification () {
+        await gateway.inject({
+          method: 'POST',
+          url: '/graphql',
+          body: {
+            query: `
+              mutation {
+                addNotification(message: "test") {
+                  id
+                }
+              }
+            `
+          }
+        })
+      }
+
+      const ws = new WebSocket(`ws://localhost:${(gateway.server.address()).port}/graphql`, 'graphql-ws')
+      const client = WebSocket.createWebSocketStream(ws, { encoding: 'utf8', objectMode: true })
+      client.setEncoding('utf8')
+      t.tearDown(() => {
+        client.destroy()
+      })
+
+      client.write(JSON.stringify({
+        type: 'connection_init',
+        payload: {
+          headers: {
+            authorize: true
+          }
+        }
+      }))
```

Comment:
Can you please add _another_ client that uses a different `authorize` header (let's use `foo` and `bar`)?

Reply:
The part you commented is not up to date but sure i'll add a different auth method

Reply:
I've added 3 authentication method: one for each client and an other for gateways. Do you want me to add a third client ?